### PR TITLE
add missing search page, properly link it

### LINKED ISSUE #48: Website search field is malfunctioning
Currently, the search field gives a 404. After a quick look, it seems to try to load a search.html page, which does not exists (AFAICS). I don't know much about the Nikola site-builder, but it seems to me the easiest/fastest solution to get search up and running, is to enable Google or DuckDuckGo search.

### PR DESCRIPTION
**Pull request by mattip**
Fixes #48 

I guess no one ever checked that search was working? Weird.

### CODE REVIEW

**Review by fangerer — APPROVED** [review]
Right, I guess no one ever tried it. Thanks, Matti.

File: conf.py [hunk 1]
```diff
@@ -676,6 +676,11 @@
 #    ".js": [filters.closure_compiler],
 #    ".jpg": ["jpegoptim --strip-all -m75 -v %s"],
 # }
+from nikola import filters
+
+FILTERS = {
+    ".html": [filters.add_header_permalinks],
+}
```
Comment (erlend-aasland) [inline conf.py:683]: This introduced a visual regression:

<details>

After this PR:
Screenshot 2024-03-22 at 00 40 21

Before this PR:
Screenshot 2024-03-22 at 00 40 43


</details>

Search works fine without the permalinks; consider removing them for less visual clutter.

Comment (hodgestar) [reply conf.py:683]: Perhaps we can just style them better? Being able to link to subheadings on pages is nice.

Comment (erlend-aasland) [reply conf.py:683]: Sure, that would be a nice solution. I'll create an issue for it, so it's not forgotten :)

**Pull request merged**